src/analysis/function_analysis.cpp

```diff
@@ -376,7 +378,8 @@ DefinednessAnalysis::DefinednessAnalysis(const ParamNames& arg_names, CFG* cfg,
     for (const auto& p : results) {
         RequiredSet required;
         for (const auto& p2 : p.second) {
-            if (scope_info->refersToGlobal(p2.first))
+            if (scope_info->refersToGlobal(p2.first)
+                || scope_info->getScopeTypeOfName(p2.first) == ScopeInfo::VarScopeType::NAME)
```

Comment:
I'm still finding it confusing to mix the old functions (refersToGlobal) and new functions (getScopeTypeOfName)... is this the same as `scope_type == GLOBAL || scope_type == NAME`?  And should the logic here be `scope_type != FAST`?